### ISSUE TITLE
Universal App template in current builds is broken

### ISSUE THREAD

**Issue by DDReaper** [mono/MonoGame#3363]
Installed the latest MG installer and tried out the Universal App template

However when building the clean project you just get the following error (for both solutions)

>Manifest references file 'UniversalGameName1.exe' which is not part of the payload.

**Comment by tomspilman** [comment]
> for both solutions

You mean both projects... Windows Phone and Windows Store...  right?

I'll have to test it here and see what is not getting expanded in the template correctly.

**Comment by DDReaper** [comment]
Yes.
Although I just tried the WP project on it's own and that is broken as well, SO I suspect it's a protobuild issue which has affected all 8.1 projects :S

**Comment by DDReaper** [comment]
Uh Oh. Just tried a WP8 project and this one's situation is worse. Can't even create that one, just get the following error:
![image](https://cloud.githubusercontent.com/assets/1793042/5633518/3536f462-95d0-11e4-9d38-e4c795f537cf.png)

**Comment by tomspilman** [comment]
This all seems fishy to me.  Are you sure you have the latest SDKs which support WP8.1 and Win8.1?

**Comment by BrianPeek** [comment]
I can +1 both of these issues. WP8 projects have been broken for awhile in MonoGame, and I just tried creating a Universal app today and get the same error as above (not part of the payload).

**Comment by BrianPeek** [comment]
For the WP8 error above, if you nav to <documents>\Visual Studio 2013\Templates\ProjectTemplates\Visual C#\MonoGame and open the WindowsPhone.zip file, you can modify the <Assembly> line to change Version=11.0.0.0 to 12.0.0.0 and have it create the project properly in VS2013.  Unfortunately I then had some reference issues and had to resort to getting the binaries from NuGet.

**Comment by BrianPeek** [comment]
Sorry for all the posts.  :)

You can fix the "not part of the payload" error by opening each Package.appxmanifest and changing the "Executable=whatever.exe" to "Executable=$targetnametoken$.exe" .

**Comment by tomspilman** [comment]
> You can fix the "not part of the payload" error by opening each Package.appxmanifest and changing the "Executable=whatever.exe" to "Executable=$targetnametoken$.exe" .

We're currently doing this...

https://github.com/mono/MonoGame/blob/develop/ProjectTemplates/VisualStudio2013/WindowsUniversal/WindowsPhone/Package-PhoneAppx.appxmanifest#L27

Is that is producing the wrong EXE?  What EXE is it generating?

Is `$targetnametoken$` part of the VS template macros or something specific to manifest?

**Comment by BrianPeek** [comment]
As far as I know, $targetnametoken$ is a VS template expansion and is what should be used here. You'll see it in any Package.appxmanifest...even the non-Universal MonoGame templates.  :)

**Comment by BrianPeek** [comment]
Let me rephrase that (after thinking about the question you asked and not what I think you asked?)...the literal $targetnametoken$ should appear in the manifest, and the build/packaging process expands this macro for the final package.  Now whether that's VS or MSBuild or what, I'm not sure.  But $targetnametoken$ is what's expected to be there...  :)

**Comment by tomspilman** [comment]
> the literal $targetnametoken$ should appear in the manifest, and the build/packaging process expands this macro for the final package.

Ok...  that makes sense.  `$ext_safeprojectname$` is a VS template macro  and I guess it isn't expanding to the right thing.

**Comment by DDReaper** [comment]
With 3.3 looming, it's imperative we fix these templates.  I can make the changes if need be (was doing some testing last night) as the fixes above to the templates seem to resolve the issues.  Will also test and verify with the individual templates as well.  
Is anyone else already working on a PR for this?

Although I did some across another bug which could be related.  If you create a new MonoGame Universal project with a space ("My Game") in the solution name, the template breaks horribly.  Does not happen with a normal Universal solution.

**Comment by BrianPeek** [comment]
I'll also add that the Universal Project doesn't setup a Content.mgcb project and associated pipeline stuffs in the csproj files if anyone's working in there.  :)  thanks!

**Comment by tomspilman** [comment]
> Is anyone else already working on a PR for this?

Not that I know of... go for it.  Just keep your changes focused on fixing the template bug.

> If you create a new MonoGame Universal project with a space ("My Game") in the solution name, the template breaks horribly.

There are like two macros...  $templatename$ and $safetemplatename$ or something like that.  Could be we're not using the "safe" one in the right places.


> I'll also add that the Universal Project doesn't setup a Content.mgcb project 

There are some complications there, so we've not supported it yet.

**Comment by DDReaper** [comment]
OK, I'll pick that up as soon as I've finished the NuGet package definitions

**Comment by DDReaper** [comment]
Submitted fixes for templates
Even managed to deploy the Content package in the Universal template
(Although did note, the ContentBuilder imports and tags were also missing from the child universal projects :S)
Tested locally, will confirm once the build process is complete

**Comment by DDReaper** [comment]
Testing complete.  Templates now behave properly and all work.
Content project and references added to Universal solution and deploys correctly
Also fixed the annoying bug of not having spaces in new project names

**Comment by BrianPeek** [comment]
Missed this comment.  Will verify this weekend!

**Comment by BrianPeek** [comment]
Ok, finally getting to this. WP80 template now works in VS, but I still get this runtime error:

{System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeLoadException: Could not find Windows Runtime type 'SharpDX.WP8.Interop'.
   at SharpDX.Direct3D11.D3D11.get_D3D11CreateDevice_()
   at SharpDX.Direct3D11.D3D11.CreateDevice(Adapter adapterRef, DriverType driverType, IntPtr software, DeviceCreationFlags flags, FeatureLevel[] featureLevelsRef, Int32 featureLevels, Int32 sDKVersion, Device deviceOut, FeatureLevel& featureLevelRef, DeviceContext& immediateContextOut)
   at SharpDX.Direct3D11.Device.GetSupportedFeatureLevel()
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.PlatformGetHighestSupportedGraphicsProfile(GraphicsDevice graphicsDevice)
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.GetHighestSupportedGraphicsProfile(GraphicsDevice graphicsDevice)
   at Microsoft.Xna.Framework.GraphicsDeviceManager..ctor(Game game)
   at Game10.Game1..ctor()
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean noCheck, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& bNeedSecurityCheck)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, StackCrawlMark& stackMark)
   at System.Activator.CreateInstance[T]()
   at MonoGame.Framework.WindowsPhone.XamlGame`1.Create(String launchParameters, PhoneApplicationPage page)
   at Game10.GamePage..ctor()
   --- End of inner exception stack trace ---
   at System.Windows.Navigation.PageResourceContentLoader.EndLoad(IAsyncResult asyncResult)
   at System.Windows.Navigation.NavigationService.ContentLoader_BeginLoad_Callback(IAsyncResult result)}

**Comment by BrianPeek** [comment]
Universal seems to work exactly as intended.  Yay!

**Comment by DDReaper** [comment]
That seems very odd on the WP8 side as I tested with both vs2012 & vs2013 (even though onlu the vs2013 templatewas updated)
So what replication steps are you using @BrianPeek ? And on which platform / editor?

**Comment by BrianPeek** [comment]
VS2013 on Win81.  Install MonoGame (latest build), open VS, File -> New Project, WP8 project, F5 to device, not emulator.

**Comment by DDReaper** [comment]
Just tested again and in the WP8 build we get this
![image](https://cloud.githubusercontent.com/assets/1793042/6070154/344f28ac-ad82-11e4-9274-4b40fb5616ed.png)
Tested it with the NuGet package (after removing 2 sets of MG references and it works fine.

So definitely a template issue

**Comment by BrianPeek** [comment]
Yup. Exactly what I'm seeing here. Setting NuGet references makes it work just fine.  Haven't had the time to explore further, unfortunately. :(

**Comment by DDReaper** [comment]
OK, I've identified the specific issue and it is just an installer issue.
raised #3574 to resolve
